src/data/reports/data-quality/DataQualityDefaultRepository.ts

```diff
@@ -0,0 +1,276 @@
+import _ from "lodash";
+import { PaginatedObjects } from "../../../domain/common/entities/PaginatedObjects";
+import {
+    IndicatorConfig,
+    IndicatorItem,
+    ProgramIndicatorConfig,
+    ProgramIndicatorItem,
+} from "../../../domain/reports/data-quality/entities/DataQualityItem";
+import {
+    DataQualityRepository,
+    IndicatorOptions,
+    ProgramIndicatorOptions,
+} from "../../../domain/reports/data-quality/repositories/DataQualityRepository";
+import { D2Api, Pager } from "../../../types/d2-api";
+import { DataStoreStorageClient } from "../../common/clients/storage/DataStoreStorageClient";
+import { StorageClient } from "../../common/clients/storage/StorageClient";
+import { Instance } from "../../common/entities/Instance";
+import { promiseMap } from "../../../utils/promises";
+import { Namespaces } from "../../common/clients/storage/Namespaces";
+
+type DataQualityItem = { [key: string]: any };
+
```

Comment:
As i understand you use this type as a catch-all for the API responses, but we have `IndicatorItem` and `ProgramIndicatorItem` for a reason.

Example taken from `DataQualityItem` usage:

```ts
// const dataQualityItem: DataQualityItem = indicator;
try {
    const numeratorValidation = await this.api.expressions
        .validate("indicator", indicator.numerator)
        .getData();
    const denominatorValidation = await this.api.expressions
        .validate("indicator", indicator.denominator)
        .getData();

    if (numeratorValidation.status === "ERROR" || denominatorValidation.status === "ERROR") {
        // !== to make false equal to invalid
        const numeratorResult = numeratorValidation.status !== "ERROR";
        const denominatorResult = denominatorValidation.status !== "ERROR";
        const metadataType = "Indicator";
        const dataQualityItem: IndicatorItem = {
            ...indicator,
            // Either this or change user type to Ref to store the User object retrived from the API
            user: indicator.user.displayName,
            numeratorresult,
            denominatorresult,
            metadataType,
        };
        dataQualityErrors.push(dataQualityItem);
    }
} catch (error) {
    console.debug(error);
}
```